source/slang/slang-ir-metal-legalize.cpp

```diff
@@ -329,8 +342,61 @@ namespace Slang
                 auto paramVarLayout = varLayoutBuilder.build();
                 builder.addLayoutDecoration(param, paramVarLayout);
             }
-        }
+            if(param->findDecorationImpl(kIROp_VerticesDecoration))
+            {
+                auto vertexRefType = (IRConstRefType*)param->getDataType();
+                auto vertexOutputType = (IRVerticesType*)vertexRefType->getValueType();
+                vertexType = vertexOutputType->getElementType();
+                maxVertices = vertexOutputType->getMaxElementCount();
+                SLANG_ASSERT(vertexType);
+
+                verticesParam = param;
+            }
+            if(param->findDecorationImpl(kIROp_IndicesDecoration))
+            {
+                auto indicesRefType = (IRConstRefType*)param->getDataType();
+                auto indicesOutputType = (IRIndicesType*)indicesRefType->getValueType();
+                indicesType = indicesOutputType->getElementType();
+                maxPrimitives = indicesOutputType->getMaxElementCount();
+                SLANG_ASSERT(indicesType);
+
+                indicesParam = param;
+            }
+            if(param->findDecorationImpl(kIROp_PrimitivesDecoration))
+            {
+                auto primitivesRefType = (IRConstRefType*)param->getDataType();
+                auto primitivesOutputType = (IRPrimitivesType*)primitivesRefType->getValueType();
+                primitiveType = primitivesOutputType->getElementType();
+                SLANG_ASSERT(primitiveType);
 
+                primitivesParam = param;
+            }
+        }
+        if(primitiveType == nullptr)
+        {
+            primitiveType = builder.getVoidType();
+        }
+        builder.setInsertBefore(entryPoint.entryPointFunc->getFirstBlock()->getFirstOrdinaryInst());
+        auto meshParam = builder.emitParam(builder.getMetalMeshType(vertexType, primitiveType, maxVertices, maxPrimitives, topology));
+        builder.addExternCppDecoration(meshParam, toSlice("_slang_mesh"));
+
+        
+        traverseUsers(verticesParam, [&](IRInst* inst){
```

Comment:
Do we need something similar for indices and primitives?

Reply:
yes, this doesn't work at all yet, I just wanted to know if this kind of approach is even reasonable